lib/ansible/modules/windows/win_zip.ps1

```diff
@@ -0,0 +1,253 @@
+#!powershell
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
```

Comment:
As @rnsc has stated we should be using the new C# Ansible.Basic util and avoiding adding anything new with the Legacy util. The benefits that Ansible.Basic gives you are

* It returns the module invocation args when you run Ansible with `-vvv`
* It easily exposes check mode, dif mode, and a temp dir that is controllable by the end user in a standard way
* It adds logging methods to log module information to the event log
* We have sanity tests that make sure the arg spec of the module matches up with the documentation
* It adds a few more option types like list and dict for better validation
* Lots more

The [win_environment](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_environment.ps1) module is a good example of how to define the module spec and other helper things like `$module.CheckMode`, `$module.Params`, `$module.Result`, `$module.FailJson()`, and finally `$module.ExitJson()`.